packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditGenerators.ts

```diff
@@ -102,56 +125,195 @@ const defaultEditGeneratorOpWeights = {
 	start: 3,
 	commit: 1,
 	abort: 1,
+	synchronize: 1,
 };
 
-export const makeEditGenerator = (
+export const makeNodeEditGenerator = (
 	opWeights: EditGeneratorOpWeights = defaultEditGeneratorOpWeights,
-): AsyncGenerator<Operation, FuzzTestState> => {
+): AsyncGenerator<NodeEdit, FuzzTestState> => {
 	type EditState = FuzzTestState & TreeContext;
-	async function insertGenerator(state: EditState): Promise<FuzzInsert> {
+
+	async function setPayloadGenerator(state: EditState): Promise<FuzzSetPayload> {
 		const trees = state.testTreeProvider.trees;
 		const tree = trees[state.treeIndex];
-
 		// generate edit for that specific tree
-		const {
-			parent: path,
-			parentField: nodeField,
-			parentIndex: nodeIndex,
-		} = getRandomPlace(tree, state.random);
-		const insert: FuzzInsert = {
-			fuzzType: "insert",
-			parent: path,
-			field: nodeField,
-			index: nodeIndex,
-			value: state.random.integer(Number.MIN_SAFE_INTEGER, Number.MIN_SAFE_INTEGER),
+		const path = getExistingRandomNodePosition(tree, state.random);
+		return {
+			nodeEditType: "setPayload",
+			path,
+			value: state.random.integer(Number.MIN_SAFE_INTEGER, Number.MAX_SAFE_INTEGER),
+			treeIndex: state.treeIndex,
 		};
-		return insert;
 	}
 
-	async function deleteGenerator(state: EditState): Promise<FuzzDelete> {
+	const baseNodeEditGenerator = createWeightedAsyncGenerator<FuzzNodeEditChange, EditState>([
+		[
+			setPayloadGenerator,
+			opWeights.setPayload ?? 0,
+			({ testTreeProvider, treeIndex }) =>
+				containsAtLeastOneNode(testTreeProvider.trees[treeIndex]),
+		],
+	]);
+
+	const buildOperation = (contents: FuzzNodeEditChange) => {
+		const operation: NodeEdit = {
+			editType: "nodeEdit",
+			edit: contents,
+		};
+		return operation;
+	};
+
+	return createAsyncGenerator<FuzzNodeEditChange, NodeEdit>(
+		baseNodeEditGenerator,
+		buildOperation,
+	);
+};
+
+export const makeFieldEditGenerator = (
+	opWeights: EditGeneratorOpWeights = defaultEditGeneratorOpWeights,
+): AsyncGenerator<FieldEdit, FuzzTestState> => {
+	type EditState = FuzzTestState & TreeContext;
+	async function fieldEditGenerator(state: EditState): Promise<SequenceFieldEdit> {
 		const trees = state.testTreeProvider.trees;
 		const tree = trees[state.treeIndex];
 		// generate edit for that specific tree
-		const { firstNode, count } = getExistingRandomNodeRangePath(tree, state.random);
-		return {
-			fuzzType: "delete",
+		const { fieldPath, fieldKey, count } = getExistingFieldPath(tree, state.random);
+		assert(fieldPath.parent !== undefined);
+		const fieldEditType = Array(opWeights.insert ?? 0)
```

Comment:
1. opWeights is determined at generator creation time, so you could do this stuff outside of the generator body. This saves some runtime.

2. Since you only have two options, you could use a `.bool()` option with appropriate weighting. If you had more, this is exactly what `createWeightedGenerator` / `createWeightedAsyncGenerator` are meant for :)

Reply:
updated to use bool()